### PR TITLE
add VPC CIDR to controller context

### PR DESCRIPTION
**Pull request by xh3b4sd**
This is one more effort to remove remote dependencies from the adapter package. A new `vpccidr` service AND resource is introduced. The resource calls a service for information and puts it into the controller context. The more resources we add to put information to the controller context, the more I think we could bundle all of these things in one resource. But that just BTW. Maybe something for later.

### CODE REVIEW

File: service/controller/v24/adapter/guest_route_tables.go [hunk 1]
```diff
@@ -19,12 +17,7 @@ type GuestRouteTablesAdapter struct {
 }
 
 func (r *GuestRouteTablesAdapter) Adapt(cfg Config) error {
-	hostClusterCIDR, err := VpcCIDR(cfg.HostClients, key.PeerID(cfg.CustomObject))
-	if err != nil {
-		return microerror.Mask(err)
-	}
```
Comment (xh3b4sd) [inline service/controller/v24/adapter/guest_route_tables.go]: Here is a call removed.

File: service/controller/v24/adapter/guest_security_groups.go [hunk 2]
```diff
@@ -42,12 +42,7 @@ type GuestSecurityGroupsAdapter struct {
 }
 
 func (s *GuestSecurityGroupsAdapter) Adapt(cfg Config) error {
-	hostClusterCIDR, err := VpcCIDR(cfg.HostClients, key.PeerID(cfg.CustomObject))
-	if err != nil {
-		return microerror.Mask(err)
-	}
```
Comment (xh3b4sd) [inline service/controller/v24/adapter/guest_security_groups.go]: Here another one.

File: service/controller/v24/adapter/guest_vpc.go [hunk 3]
```diff
@@ -58,25 +57,3 @@ func (v *GuestVPCAdapter) Adapt(cfg Config) error {
 
 	return nil
 }
-
-func VpcCIDR(clients Clients, vpcID string) (string, error) {
```
Comment (xh3b4sd) [inline service/controller/v24/adapter/guest_vpc.go]: Not needed anymore.

File: service/controller/v24/cluster_resource_set.go [hunk 4]
```diff
@@ -561,6 +575,7 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 
 	resources := []controller.Resource{
 		accountIDResource,
+		vpcCIDRResource,
```
Comment (xh3b4sd) [inline service/controller/v24/cluster_resource_set.go]: Wiring the new resource.

File: service/controller/v24/resource/tccp/main_stack.go [hunk 5]
```diff
@@ -36,14 +36,15 @@ func (r *Resource) getMainGuestTemplateBody(ctx context.Context, customObject v1
...
 		},
-		CustomObject:      customObject,
-		Clients:           adapterClients,
-		EncrypterBackend:  r.encrypterBackend,
-		HostClients:       *r.hostClients,
-		InstallationName:  r.installationName,
-		HostAccountID:     hostAccountID,
-		PublicRouteTables: r.publicRouteTables,
-		Route53Enabled:    r.route53Enabled,
+		ControlPlaneVPCCidr: cc.Status.ControlPlane.VPC.CIDR,
```
Comment (xh3b4sd) [inline service/controller/v24/resource/tccp/main_stack.go]: Here is the CIDR taken from the context and injected into the adapter.

File: service/controller/v24/resource/vpccidr/resource.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+			return microerror.Mask(err)
+		}
+	}
+
+	cidr, err := vpcCIDRService.Lookup(ctx, key.PeerID(cr))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	cc.Status.ControlPlane.VPC.CIDR = cidr
```
Comment (xh3b4sd) [inline service/controller/v24/resource/vpccidr/resource.go]: Here is the resource magic.

File: service/vpccidr/vpc_cidr.go [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+type Config struct {
+	EC2    EC2
+	Logger micrologger.Logger
+}
+
+type VPCCIDR struct {
```
Comment (xh3b4sd) [inline service/vpccidr/vpc_cidr.go]: This is the new service implementation.

**Comment by xh3b4sd** [comment]
Works like a charm. Tested on `gauss`.

**Comment by xh3b4sd** [comment]
Fishing LGTM.

File: service/vpccidr/vpc_cidr.go [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+	ec2    EC2
+	logger micrologger.Logger
+
+	// cidrs is a mapping of vpcs IDs and CIDRs, where the key is the ID
+	// and the value is the CIDR.
+	cidrs map[string]string
```
Comment (kopiczko) [inline service/vpccidr/vpc_cidr.go]: I wonder why do we cache results and put them in the context. Isn't it ok to call this service from tccp directly?

Comment (xh3b4sd) [reply service/vpccidr/vpc_cidr.go]: The caching is indeed rather useless right now. I put the information into the context to have it generally available. I still need to wrap my head around where we need which information and which information we need additionally for node pools. Will figure this out.

**Pull request merged**